### DISCUSSION TITLE
What is Startup Templates

### DISCUSSION THREAD

**Discussion by samandar-boymurodov** [SilentVoid13/Templater#1270] [Help]
Hello Templater team,
I would like to know how Startup Templates feature works in Templater plugin.
![image](https://github.com/SilentVoid13/Templater/assets/65066111/405398d9-68e7-4daa-869b-14d7a1891a86)

From its description, it feels that we can run specific hooks on note creation, deletion, and so on. But I could not actually use it. Is there any guide on how to use it properly?

**Comment by Zachatoo** [comment]
A startup template is a template that will run when Obsidian is opened. That's it. You could have a template like the following, and every time you open Obsidian, you'll get a notification.

```js
<%* new Notice("Startup complete!") -%>
```

Now, usually this isn't very useful unless you're setting up global variables you want to use in other scripts, or to hook into Obsidian events. You can look at the [Obsidian API docs](https://docs.obsidian.md/Home) to find a what events you can hook into. For example, if you want to hook into the [file create](https://docs.obsidian.md/Reference/TypeScript+API/Vault/on) event, you can do something like this.

```js
<%*
app.vault.on("create", (file) => {
	new Notice(`File was created with name '${file.basename}'`);
});
-%>
```
